src/pages/resultsView/querySummary/QuerySummary.tsx

```diff
@@ -77,8 +119,37 @@ export default class QuerySummary extends React.Component<{ routingStore:Extende
                  ><i className="fa fa-info-circle"/>
                 </DefaultTooltip>
             </span>
-        </div>
-    }
+        </div>)
+    });
+
+    readonly cohortAndGeneSummary = remoteData({
+        await:()=>[this.singleStudyUI, this.multipleStudyUI, this.props.store.queriedStudies],
+        invoke:()=>{
+            if (this.props.store.queriedStudies.result.length === 1) {
+                return Promise.resolve(this.singleStudyUI.result!);
+            } else {
+                return Promise.resolve(this.multipleStudyUI.result!);
+            }
+        }
+    });
+
+    readonly alterationSummary = remoteData({
+        await:()=>[this.props.store.samples, this.props.store.patients,
+            this.props.store.alteredSampleKeys, this.props.store.alteredPatientKeys],
+        invoke:()=>{
+            const numSamples = this.props.store.samples.result.length;
+            const numPatients = this.props.store.patients.result.length;
+            const numAlteredSamples = this.props.store.alteredSampleKeys.result!.length;
+            const numAlteredPatients = this.props.store.alteredPatientKeys.result!.length;
+            const sampleSummary = `${numSamples} (${getPercentage(numAlteredSamples/numSamples, 0)}) of queried samples`;
```

Comment:
it think this should all be in pure functions (construction of the string)

Reply:
I dont see the point here, bc its only one line and only called once

Reply:
sure :+1: